setup.py

```diff
@@ -126,13 +148,15 @@ def build_extension(self, ext):
                 ext.include_dirs.append(LOGGING_SRC_DIR)
 
                 if sys.platform == "win32":
+                    if not any("/std" not in s for s in ext.extra_compile_args):
+                        ext.extra_compile_args.append("/std:c++17")
```

Comment:
It looks like the c++ standard has to be set on Windows now too @xhochy and @3dbrows

Reply:
Yes, Arrow 10 switched to C++17 as the minimal supported C++ standard.